10up/ads-txt
# End-to-end testing with Cypress

### Description of the Change




Closes #79 

Added end-to-end testing with Cypress. Runs automatically on every pull request to `develop` branch.

Steps to perform local testing or develop new tests:
1. Install dependencies `npm install`
2. Start wp-env local development suite `npm run env:start`
3. Run cypress `npm run cypress:run` or open developer browser `npm run cypress:open`

Includes ./tests/bin/set-core-version.js to change WordPress version for wp-env. Usage:

```bash
./tests/bin/set-core-version.js WordPress/WordPress#5.2
```

**Testing environment:** WordPress latest

**Testing suite:**
- Activate plugin
- Check pretty links are enabled
- Open Ads.txt settings page
- Save invalid record expecting error message
- Save valid record
- Read and check http://localhost/ads.txt
- Test revisions
- Open App-ads.txt settings page
- Save invalid record expecting error message
- Save valid record
- Read and check http://localhost/app-ads.txt
- Test revisions
 
### Verification Process

GitHub Actions workflow "End-to-end Testing", expected to pass the test. Refer to "Description of the Change" for local testing.

### Checklist:






### Changelog Entry

> Add - End-to-end testing with Cypress

### Credits


Props @cadic, @dinhtungdu, @darylldoyle, @Sidsector9
